RxCocoa/Common/DelegateProxy.swift

```diff
@@ -142,17 +143,18 @@ open class DelegateProxy : _RXDelegateProxy {
     - returns: Observable sequence of arguments passed to `selector` method.
      */
     open func methodInvoked(_ selector: Selector) -> Observable<[Any]> {
+        MainScheduler.ensureExecutingOnScheduler()
```

Comment:
Hi @sergdort , 

If `DelegateProxyType` interface would allow concurrent usage then we would also need it to provide some sort of synchronization mechanism that all extension methods would need to use.

That synchronization mechanism would also need to be native to the delegate owner.

So far 99% of classes that have delegates and are using delegate proxy are UI elements that use main dispatch queue as a synchronization mechanism.

The only case I know of that can use other synchronization mechanism is `CBCentralManager`. You can pass a specific dispatch queue to it's initializer but it's not exposed anywhere on it and I'm not exactly sure what restrictions does it create if you do.

This API only restricts calling `methodInvoked` from main scheduler and not subscribing to it.

As far as I can tell this is the simplest API that works all use cases and I'm not sure what would be the best way to make it thread safe and what benefit we would get from it.